### ISSUE TITLE
LDAP/AD login fails when AD property contains binary blob

### ISSUE THREAD

**Issue by S1M0N** [Requarks/wiki#849]
**Describe the bug**
When attempting to perform login using LDAP/AD the login fails with what appears to be a DB related issue. Issue comes from what appears to be a PostgreSQL limitation of it not accepting NULL (\0x00) characters in text fields.

Error Received: 
```
insert into "users" ("createdAt", "email", "isActive", "isSystem", "isVerified", "localeCode", "name", "pictureUrl", "providerId", "providerKey", "tfaIsActive", "updatedAt") values ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12) returning "id" - invalid byte sequence for encoding "UTF8": 0x00
```
The below section of this error points to PostgreSQL not supporting the storage of NULL (\0x00) characters in text fields.
```
invalid byte sequence for encoding "UTF8": 0x00
```

**To Reproduce**
1. Enable LDAP/AD Authentication Method and enter appropriate LDAP/AD Data to allow authentication.
2. Enable "Allow Self-Registration" for LDAP/AD Authentication (without this user receives "You are not authorized to login." message).
2. Open incognito window and attempt to login with known good LDAP User Credentials. 
3. Receive above error.

**Expected behavior**
User able to login with no errors. If user is not in DB then Create User process happens without error and user is logged in.

**Screenshots**
![image](https://user-images.githubusercontent.com/727018/58129422-b07aa400-7be7-11e9-9e03-ee662011f1e9.png)

**Host Info (please complete the following information):**
 - OS: Ubuntu 18.04 LTS
 - Wiki.js version: 2.0.0-beta.115
 - Database engine: PostgreSQL 10.8 (Ubuntu 10.8-0ubuntu0.18.04.1) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 7.3.0-16ubuntu3) 7.3.0, 64-bit

**Additional context**
Secondary to the above, I am not able to create new users manually in the admin panel, as the option is greyed out. 
![image](https://user-images.githubusercontent.com/727018/58129881-c2107b80-7be8-11e9-980f-5280db5c2dd8.png)

**Comment by NGPixel** [comment]
Can you try mapping the avatar field to another AD property (e.g. `mail`)?

It seems like one of the fields is returning binary data.

**Comment by S1M0N** [comment]
That seems to have done the trick, changed that field to `entryUUID` since it's unique to each user, and the test user is now able to login successfully. They appear in the user list as expected, and are assigned the appropriate group as designated in the admin panel. 

Any thoughts on why I can't generate a new user locally?

**Comment by NGPixel** [comment]
That feature isn't working yet. It will become available in a future beta build. For now you must use the self-registration technique to create new users.

For the avatar issue, I'll mark this as a bug as other users will likely encounter the same issue. A filter should be applied on that field.

**Comment by S1M0N** [comment]
Gotcha, much appreciated for the time and workaround!

**Comment by col-panic** [comment]
Confirming bug with an avatar image set, version is 2.0.12
```
2020-01-16T09:55:37.543Z [MASTER] warn: LDAP LOGIN ERROR (c2): insert into `users` (`createdAt`, `email`, `isActive`, `isSystem`, `isVerified`, `localeCode`, `name`, `pictureUrl`, `providerId`, `providerKey`, `tfaIsActive`, `updatedAt`) values ('2020-01-16T09:55:37.540Z', 'redacted@redacted.com', true, false, true, 'de', 'My Name', '����\0JFIF\0\0\0\0\0��\0C\0\b\t\b\n\n\t\b\t\t\n

                                                                                                                                \n

                                                                                                                                  \t\t\r\r\n
                                                                                                                                            ��\0C\b\b
                                                                                                                                                     \t
                                                                                                                                                       ��\0\b\0�\0�\"\0��\00\0\0\0\0\0\0\0\0\0\0\0\0\b��\0D\0\t\0\0\0\0!1A\"Qa2q�B���\b#3R����bcr���&Cs������\00\0\0\0\0\0\0\0\0\0\0\0��\0.\0\0\0\0\0\0\0\0!1A\"aq2Q�B��#3�����\0
```

**Comment by kingfisher77** [comment]
Same here with attribute `jpegPoto` from Samba AD

At login:
`update "users" set "email" = $1, "name" = $2, "pictureUrl" = $3, "updatedAt" = $4 where "users"."id" = $5 - invalid byte sequence for encoding "UTF8": 0x00
`

Deleting the photo in the directory enables login again.

**Comment by Turakar** [comment]
Same as @kingfisher77 with PostgreSQL 11, wikijs 2.2.51 and OpenLDAP

**Comment by Hasztagg** [comment]
Can confirm @Turakar, we are able to replicate that in Wiki.js 2.2.51 with Active Directory for a user that had `jpegPhoto` set.

Here's a summary of that temporary workaround:
1. Go to wiki/a/auth page.
2. Set *Avatar Picture Field Mapping* to something else than `jpegPhoto` (note: this will work for all existing and future users). We set it to `entryUUID` and it works fine.
3. Users with this issue will now be able to log in (or self-register) normally.

Not sure what's that field used for anyway in 2.2.51, but that shouldn't introduce any new issues.

**Comment by NGPixel** [comment]
Fixed in **2.5.126**
